File: frontend/src/app/core/services/cuenta-por-cobrar.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { environment } from '../../../environments/environment';
import {
  CuentaPorCobrar,
  GenerarPuestosConsumoRequest,
  GenerarPuestosMontoFijoRequest,
  GenerarSociosRequest,
} from '../models/cuenta-por-cobrar.model';

@Injectable({ providedIn: 'root' })
export class CuentaPorCobrarService {
  private readonly baseUrl = `${environment.apiUrl}/cuentas-por-cobrar`;

  constructor(private readonly http: HttpClient) {}

  listar(): Observable<CuentaPorCobrar[]> {
    return this.http.get<CuentaPorCobrar[]>(this.baseUrl);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }

  generarParaPuestosMontoFijo(request: GenerarPuestosMontoFijoRequest): Observable<CuentaPorCobrar[]> {
    return this.http.post<CuentaPorCobrar[]>(`${this.baseUrl}/generar/puestos-monto-fijo`, request);
  }

  generarParaPuestosConsumo(request: GenerarPuestosConsumoRequest): Observable<CuentaPorCobrar[]> {
    return this.http.post<CuentaPorCobrar[]>(`${this.baseUrl}/generar/puestos-consumo`, request);
  }

  generarParaSocios(request: GenerarSociosRequest): Observable<CuentaPorCobrar[]> {
    return this.http.post<CuentaPorCobrar[]>(`${this.baseUrl}/generar/socios`, request);
  }
}

File: backend/src/main/java/com/cibertec/sigac/service/CuentaPorCobrarService.java
package com.cibertec.sigac.service;

import java.util.List;

import com.cibertec.sigac.dto.CuentaPorCobrarResponse;
import com.cibertec.sigac.dto.GenerarPuestosConsumoRequest;
import com.cibertec.sigac.dto.GenerarPuestosMontoFijoRequest;
import com.cibertec.sigac.dto.GenerarSociosRequest;

public interface CuentaPorCobrarService {

    List<CuentaPorCobrarResponse> listarTodas();

    CuentaPorCobrarResponse obtenerPorId(Long id);

    void eliminar(Long id);

    List<CuentaPorCobrarResponse> generarParaPuestosMontoFijo(GenerarPuestosMontoFijoRequest request);

    List<CuentaPorCobrarResponse> generarParaPuestosConsumo(GenerarPuestosConsumoRequest request);

    List<CuentaPorCobrarResponse> generarParaSocios(GenerarSociosRequest request);
}
